allennlp/models/semantic_parsing/text2sql_parser.py

```diff
@@ -0,0 +1,399 @@
+import logging
```

Comment:
So, this model looks good to me, but here's a higher-level question:

There is almost nothing in here that is specific to SQL - this is a standard grammar-constrained semantic parser.  It's basically identical to the NLVR model, where the only real difference is in the executor and the metric.  Is there a way to have a model implemented that people can just use, with ~10 lines of code (for the execution and metric bits) instead of ~400?

Reply:
Yes, but I don't want to think about that until after the NACCL deadline, because there are going to have to be quite a few modifications to this to run the ablations etc.

Reply:
Yeah, I agree, not for this PR.  Just something on my mind.